### ISSUE TITLE
Implement overall pretraining -> downstream adaptation workflow in TopoBench

### ISSUE THREAD

**Issue by LouisVanLangendonck** [LouisVanLangendonck/TopoBench_challenge#12]
- Define pre-training logic in topobench via: 
-- Custom Wrappers
-- Custom Readouts
-- Custom Loss
-- Custom Evaluator
-- Optionally a Custom Pre-Transform

- Run Downstream Adaptation (either inductive or transductive):
-- Select pretrained run DIRECTORY in wandb file OR when using grid script, giving wandb project name and tries all
-- For inductive either community-related tasks (Community Detection, Homophily, Graph-level Community Detection) OR graph-property tasks (Diameter, avg degree)
-- For transductive only community-detection task!
-- Few shot learning!
